src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm2/Controllers/TeamPlanController/TeamPlanController.spec.tsx

```diff
@@ -0,0 +1,336 @@
+//     'Unlimited public repositories',
+//     'Unlimited private repositories',
+//     'Priority Support',
+//   ],
+//   monthlyUploadLimit: null,
+//   quantity: 13,
+// }
+
+const teamPlanMonth = {
+  baseUnitPrice: 5,
+  benefits: ['Up to 10 users'],
+  billingRate: 'monthly',
+  marketingName: 'Users Team',
+  monthlyUploadLimit: 2500,
+  value: 'users-teamm',
+  quantity: 10,
+}
+
+const teamPlanYear = {
+  baseUnitPrice: 4,
+  benefits: ['Up to 10 users'],
+  billingRate: 'annually',
+  marketingName: 'Users Team',
+  monthlyUploadLimit: 2500,
+  value: 'users-teamy',
+  quantity: 5,
+}
+
+const mockAccountDetailsBasic = {
+  plan: basicPlan,
+  activatedUserCount: 1,
+  inactiveUserCount: 0,
+}
+
+const mockAccountDetailsTeamMonthly = {
+  plan: teamPlanMonth,
+  activatedUserCount: 7,
+  inactiveUserCount: 0,
+  subscriptionDetail: {
+    latestInvoice: {
+      periodStart: 1595270468,
+      periodEnd: 1597948868,
+      dueDate: '1600544863',
+      amountPaid: 9600.0,
+      amountDue: 9600.0,
+      amountRemaining: 0.0,
+      total: 9600.0,
+      subtotal: 9600.0,
+      invoicePdf:
+        'https://pay.stripe.com/invoice/acct_14SJTOGlVGuVgOrk/invst_Hs2qfFwArnp6AMjWPlwtyqqszoBzO3q/pdf',
+    },
+  },
+}
+
+const mockAccountDetailsTeamYearly = {
+  plan: teamPlanYear,
+  activatedUserCount: 11,
+  inactiveUserCount: 0,
+}
+
+const mockPlanDataResponse = {
+  baseUnitPrice: 10,
+  benefits: [],
+  billingRate: 'monthly',
+  marketingName: 'Pro Team',
+  monthlyUploadLimit: 2500,
+  value: 'test-plan',
+  trialStatus: TrialStatuses.NOT_STARTED,
+  trialStartDate: '',
+  trialEndDate: '',
+  trialTotalDays: 0,
+  pretrialUsersCount: 0,
+  planUserCount: 1,
+}
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      suspense: true,
+      retry: false,
+    },
+  },
+  logger: {
+    error: () => null,
+    warn: () => null,
+    log: () => null,
+  },
+})
+const server = setupServer()
+
+beforeAll(() => server.listen())
+afterEach(() => {
+  queryClient.clear()
+  server.resetHandlers()
+})
+afterAll(() => server.close())
+
+type WrapperClosure = (
+  initialEntries?: string[]
+) => React.FC<React.PropsWithChildren>
+const wrapper: WrapperClosure =
+  (initialEntries = ['/gh/codecov']) =>
+  ({ children }) =>
+    (
+      <QueryClientProvider client={queryClient}>
+        <MemoryRouter initialEntries={initialEntries}>
+          <Route path="/:provider/:owner">
+            <Suspense fallback={null}>{children}</Suspense>
+          </Route>
+        </MemoryRouter>
+      </QueryClientProvider>
+    )
+
+interface SetupArgs {
+  planValue: string
+  errorDetails?: string
+}
+
+describe('TeamPlanController', () => {
+  function setup(
+    { planValue = Plans.USERS_BASIC }: SetupArgs = {
+      planValue: Plans.USERS_BASIC,
+    }
+  ) {
+    const addNotification = jest.fn()
+    const user = userEvent.setup()
+
+    //@ts-ignore
+    useAddNotification.mockReturnValue(addNotification)
```

Comment:
To get rid of the `@ts-ignore` you can follow the following, and also have it be "typesafe" at least in the file: https://github.com/codecov/gazebo/blob/main/src/pages/PullRequestPage/subroute/ComponentsSelector/ComponentsSelector.spec.tsx#L14-L15